spec/classes/openvpn_config_spec.rb

```diff
@@ -1,37 +1,41 @@
 require 'spec_helper'
 
 describe 'openvpn::config', type: :class do
-  context 'on Debian based machines' do
-    let(:facts) do
-      {
-        osfamily: 'Debian',
-        operatingsystem: 'Debian',
-        operatingsystemrelease: '7',
-        concat_basedir: '/var/lib/puppet/concat'
-      }
-    end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts.merge(
+          concat_basedir: '/var/lib/puppet/concat'
```

Comment:
Did you test if this is required? We have mocked facts here: https://github.com/voxpupuli/puppet-openvpn/blob/9f3abf684f9d817e67a56469c15ea4028eae34d7/spec/default_facts.yml#L11

And use them here:
https://github.com/voxpupuli/puppet-openvpn/blob/9f3abf684f9d817e67a56469c15ea4028eae34d7/spec/spec_helper.rb#L25-L35